enclave-runtime/src/tests.rs

```diff
@@ -127,648 +131,313 @@ pub extern "C" fn test_main_entrance() -> size_t {
-	// ensure state starts empty
-	state::init_shard(&shard).unwrap();
-	let mut state = Stf::init_state();
+	let (_, mut state, shard, _) = test_setup();
+
+	let signed_top_hashes: Vec<H256> = vec![[94; 32].into(), [1; 32].into()].to_vec();
+
 	Stf::update_sidechain_block_number(&mut state, 3);
 
 	// when
 	let (opaque_call, signed_block) = crate::compose_block_and_confirmation::<Block, SignedBlock>(
-		&latest_onchain_header,
+		&latest_parentchain_header(),
 		signed_top_hashes,
 		shard,
-		state_hash_apriori,
+		state.hash(),
 		&mut state,
 	)
 	.unwrap();
-	let xt_block_encoded = [TEEREX_MODULE, BLOCK_CONFIRMED].encode();
-	let block_hash_encoded = blake2_256(&signed_block.block().encode()).encode();
-	let mut opaque_call_vec = opaque_call.encode();
 
 	// then
+	let expected_call = OpaqueCall::from_tuple(&(
+		[TEEREX_MODULE, BLOCK_CONFIRMED],
+		shard,
+		blake2_256(&signed_block.block().encode()),
+	));
+
 	assert!(signed_block.verify_signature());
 	assert_eq!(signed_block.block().block_number(), 4);
-	assert!(opaque_call_vec.starts_with(&xt_block_encoded));
-	let mut stripped_opaque_call = opaque_call_vec.split_off(xt_block_encoded.len());
-	assert!(stripped_opaque_call.starts_with(&shard.encode()));
-	let stripped_opaque_call = stripped_opaque_call.split_off(shard.encode().len());
-	assert!(stripped_opaque_call.starts_with(&block_hash_encoded));
+	assert!(opaque_call.encode().starts_with(&expected_call.encode()));
 
 	// clean up
 	state::tests::remove_shard_dir(&shard);
 }
 
-#[allow(unused)]
 fn test_submit_trusted_call_to_top_pool() {
 	// given
-	ensure_no_empty_shard_directory_exists();
+	let (top_pool, _, shard, mrenclave) = test_setup();
 
-	// create top pool
-	let api: Arc<SideChainApi<Block>> = Arc::new(SideChainApi::new());
-	let tx_pool = BasicPool::create(Default::default(), api, Arc::new(TestRpcResponder::new()));
-	let author = Author::new(Arc::new(&tx_pool));
+	let sender = funded_pair();
 
-	// create trusted call signed
-	let nonce = 0;
-	let mrenclave = [0u8; 32];
-	let shard = ShardIdentifier::default();
-	// ensure state starts empty
-	state::init_shard(&shard).unwrap();
-	Stf::init_state();
-	let signer_pair = Ed25519Seal::unseal().unwrap();
-	let call = TrustedCall::balance_set_balance(
-		signer_pair.public().into(),
-		signer_pair.public().into(),
-		42,
-		42,
-	);
-	let signed_call = call.sign(&signer_pair.into(), nonce, &mrenclave, &shard);
-	let trusted_operation: TrustedOperation = signed_call.clone().into_trusted_operation(true);
-	// encrypt call
-	let rsa_pubkey = Rsa3072Seal::unseal_pubkey().unwrap();
-	let mut encrypted_top: Vec<u8> = Vec::new();
-	rsa_pubkey
-		.encrypt_buffer(&trusted_operation.encode(), &mut encrypted_top)
-		.unwrap();
+	let signed_call =
+		TrustedCall::balance_set_balance(sender.public().into(), sender.public().into(), 42, 42)
+			.sign(&sender.into(), 0, &mrenclave, &shard);
 
 	// when
+	submit_top_to_top_pool(&top_pool, direct_top(signed_call.clone()), shard);
```

Comment:
Is it really okay to move the to be tested function to an outsourced function? 
The first idea I get when looking at  this test is that the function `submit_top_to_top_pool` is tested. But that's not the case. This kind of makes the `//when` missleading

Reply:
Oh, you are looking closely. Good. I also thought about that. But when reading the test's name. I think this is exactly what is being tested.

Actually, I believe it is even good that `submit_top_to_top_pool` is tested. Test functions should also be tested, so we can make sure they do what they should in other tests.